cli/src/pcluster/aws/common.py

```diff
@@ -184,6 +184,12 @@ def clear_all():
         for cache in Cache._caches:
             cache.clear()
 
+    @staticmethod
+    def add_cache(custom_cache):
```

Comment:
let's add typing to clarify which kind of items we're adding.

Reply:
After chatting with demartinofra, we moved away from this logic. Then we added a `reset() in AWSApi class to reset the singleton to None when caches need to be cleared